pkg/vm/cli/cli.go

```diff
@@ -1,197 +1,316 @@
+	c.Printf("instruction pointer at %d (%s)\n", ip, opcode)
+}
+
+func handleBreak(c *ishell.Context) {
+	if (!checkVMIsReady(c)) {
 		return
 	}
-	if com.ready && !c.vm.Ready() {
-		fmt.Println("VM is not ready: no program loaded")
+	v := getVMFromContext(c)
+	if len(c.Args) != 1 {
+		c.Err(errors.New("Missing parameter <ip>"))
+	}
+	n, err := strconv.Atoi(c.Args[0])
+	if err != nil {
+		c.Err(errors.New(fmt.Sprintf("argument conversion error: %s\n", err)))
 		return
 	}
 
-	switch cmd {
-	case "help":
-		printHelp()
-
-	case "exit":
-		fmt.Println("Bye!")
-		os.Exit(0)
+	v.AddBreakPoint(n)
+	c.Printf("breakpoint added at instruction %d\n", n)
+}
 
-	case "ip":
-		ip, opcode := c.vm.Context().CurrInstr()
-		fmt.Printf("instruction pointer at %d (%s)\n", ip, opcode)
+func handleXStack(c *ishell.Context) {
+	v := getVMFromContext(c)
+	c.Println(v.Stack(c.Cmd.Name))
+}
 
-	case "break":
-		n, err := strconv.Atoi(args[0])
-		if err != nil {
-			fmt.Printf("argument conversion error: %s\n", err)
-			return
-		}
+func handleLoadAVM(c *ishell.Context) {
+	v := getVMFromContext(c)
+	if err := v.LoadFile(c.Args[0]); err != nil {
+		c.Err(err)
+	} else {
+		c.Printf("READY: loaded %d instructions\n", v.Context().LenInstr())
+	}
+	changePrompt(c, v)
+}
 
-		c.vm.AddBreakPoint(n)
-		fmt.Printf("breakpoint added at instruction %d\n", n)
+func handleLoadHex(c *ishell.Context) {
+	v := getVMFromContext(c)
+	b, err := hex.DecodeString(c.Args[0])
+	if err != nil {
+		c.Err(err)
+		return
+	}
+	v.Load(b)
+	c.Printf("READY: loaded %d instructions\n", v.Context().LenInstr())
+	changePrompt(c, v)
+}
 
-	case "estack", "istack", "astack":
-		fmt.Println(c.vm.Stack(cmd))
+func handleLoadGo(c *ishell.Context) {
+	v := getVMFromContext(c)
+	fb, err := ioutil.ReadFile(c.Args[0])
+	if err != nil {
+		c.Err(err)
+		return
+	}
+	b, err := compiler.Compile(bytes.NewReader(fb), &compiler.Options{})
+	if err != nil {
+		c.Err(err)
+		return
+	}
 
-	case "loadavm":
-		if err := c.vm.LoadFile(args[0]); err != nil {
-			fmt.Println(err)
-		} else {
-			fmt.Printf("READY: loaded %d instructions\n", c.vm.Context().LenInstr())
-		}
+	v.Load(b)
+	c.Printf("READY: loaded %d instructions\n", v.Context().LenInstr())
+	changePrompt(c, v)
+}
 
-	case "loadhex":
-		b, err := hex.DecodeString(args[0])
+func handleRun(c *ishell.Context) {
+	v := getVMFromContext(c)
+	if len(c.Args) != 0 {
+		var (
+			method []byte
+			params []vm.StackItem
+			err    error
+		)
+		method = []byte(c.Args[0])
+		params, err = parseArgs(c.Args[1:])
 		if err != nil {
-			fmt.Println(err)
+			c.Err(err)
 			return
 		}
-		c.vm.Load(b)
-		fmt.Printf("READY: loaded %d instructions\n", c.vm.Context().LenInstr())
+		v.LoadArgs(method, params)
+	}
+	v.Run()
+	changePrompt(c, v)
+}
 
-	case "loadgo":
-		fb, err := ioutil.ReadFile(args[0])
-		if err != nil {
-			fmt.Println(err)
-			return
-		}
-		b, err := compiler.Compile(bytes.NewReader(fb), &compiler.Options{})
+func handleCont(c *ishell.Context) {
```

Comment:
nit: handleCont maybe handleContext or handleCtx since we haven't used in the code 'Cont' short naming for the context

Reply:
It actually is `cont` because the command itself is named `cont` from `continue`. And all `handle*` are named like that, so it's a feature.

Reply:
now I got it =)